Android/MMKV/mmkv/src/main/cpp/MMKV.cpp

```diff
@@ -1204,11 +1211,48 @@ bool MMKV::isFileValid(const std::string &mmapID) {
     }
 }
 
+static void mkSpecialCharacterFileDirectory() {
+    char *path = strdup((g_rootDir + "/" + SPECIAL_CHARACTER_DIRECTORY_NAME).c_str());
+    mkPath(path);
+    free(path);
+}
+
+static void md5(string &value) {
+    unsigned char md[MD5_DIGEST_LENGTH];
+    char tmp[3] = {'\0'}, buf[33] = {'\0'};
+    MD5((const unsigned char *)value.c_str(), value.size(), md);
+    for(int i = 0;i < MD5_DIGEST_LENGTH;i++)
+    {
+        sprintf(tmp, "%2.2x", md[i]);
+        strcat(buf, tmp);
+    }
+    value = buf;
+}
+
+static string encodeFilePath(const string &mmapID) {
+    const char *specialCharacters = "\\/:*?\"<>|";
+    string filePath(mmapID);
```

Comment:
This string copy is unnecessary. I suggest making void md5(string&) into string md5(const string&). And filePath = md5(mmapID).

Reply:
@lingol 👍 I removed copy of constructor.